<a href="https://colab.research.google.com/github/Fernando-cfs/projetomodulo4/blob/main/individualmodulo4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#<font color = red> Projeto Individual Modulo 4

## Sobre o Negócio

A empresa 'Soft' de desenvolvimento de software quer expandir no mercado global para isso, quer saber como anda o desempenho dos seus funcionários e como pode melhorar para ter maior proveito deles e com isso maior satisfação dos clientes para eles se torna uma referência em sua área para isso será gerado insights em relação aos seguintes tópicos: Dia,	Horas Trabalhadas,	Bugs Corrigidos,	Tarefas Concluídas. Com essas informações teremos insights de como anda o desempenhos dos funcionários da 'Soft' e como melhora-los



## Importando as bibliotecas
- Pandas
- plotly.express
- plotly.graph_objects

In [39]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go


## Criação das tabelas
- Dia
- Horas Trabalhadas
- Bugs Corrigidos
- Tarefas concluídas

In [40]:
tabelas = {'Dia': ['Segunda','Terça', 'Quarta','Quinta','Sexta','Sábado', 'Domingo'],
         'Horas Trabalhadas': [6,7,8,6,7,5,4],
              'Bugs Corrigidos': [3,2,1,4,3,2,1],
              'Tarefas Concluídas': [5,4,6,4,5,3,2]}

dftabelas = pd.DataFrame(tabelas)
dftabelas

Dia  Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas
0  Segunda                  6                3                   5
1    Terça                  7                2                   4
2   Quarta                  8                1                   6
3   Quinta                  6                4                   4
4    Sexta                  7                3                   5
5   Sábado                  5                2                   3
6  Domingo                  4                1                   2

In [41]:
# Estatistica descritivas sobre a tabela
dftabelas.describe()

Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas
count           7.000000         7.000000            7.000000
mean            6.142857         2.285714            4.142857
std             1.345185         1.112697            1.345185
min             4.000000         1.000000            2.000000
25%             5.500000         1.500000            3.500000
50%             6.000000         2.000000            4.000000
75%             7.000000         3.000000            5.000000
max             8.000000         4.000000            6.000000

Na análise descritiva do desempenho dos funcionários de uma empresa podemos observar que 75% das horas trabalhadas esta a baixo de 7 horas e que a minima de horas trabalhadas e de 4 horas e a máxima e de 8 horas


In [42]:
# correlacao
dftabelas.corr(numeric_only=True)

Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas
Horas Trabalhadas            1.000000         0.079536            0.907895
Bugs Corrigidos              0.079536         1.000000            0.190885
Tarefas Concluídas           0.907895         0.190885            1.000000

Na correlação entre bug e horas trabalhas não é muito grande, tem uma correlação maior entre horas trabalhas e tarefas concluídas


## Criação dos cálculos média diaria, total e gráficos


- Total de Horas Trabalhadas
- Média Diária de Horas Trabalhadas

In [43]:

# Criar DataFrame a partir das tabelas
horas = pd.DataFrame(tabelas)

# Calcular a média e a soma total das horas trabalhadas
media_horas = horas['Horas Trabalhadas'].mean()
total_horas = horas['Horas Trabalhadas'].sum()

# Criar figura com os dados
fig = go.Figure()

# Adicionar as barras para média e total e configuralas
fig.add_trace(go.Bar(x=['Média Diária', 'Total por Semana'],
                     y=[media_horas, total_horas],
                     marker=dict(color=['green', 'red']),
                     text=[f'Média: {media_horas:.2f} horas', f'Total: {total_horas:.2f} horas'],
                     textposition='auto'
                    ))
# Personalizar o layout do gráfico
fig.update_layout(title='Média e Total de Horas Trabalhadas por Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Estatísticas',
                  yaxis_title='Horas',
                  showlegend=False,
                  width=600  # Definindo a largura do gráfico para 600 pixels
                  )
# Mostrar o gráfico interativo
fig.show()


-------------------------------------------------------------------------------

- Total de Bugs Corrigidos
- Média Diária de Bugs Corrigidos

In [44]:
# Criar DataFrame a partir das tabelas
bgcorrigidos = pd.DataFrame(tabelas)

# Calcular a média e a soma total dos Bugs Corrigidos
media_bgcorrigidos = horas['Bugs Corrigidos'].mean()
total_bgcorrigidos = horas['Bugs Corrigidos'].sum()

# Criar figura com os dados
fig = go.Figure()

# Adicionar as barras para média e total e configuralas
fig.add_trace(go.Bar(x=['Média Diária', 'Total por Semana'],
                     y=[media_bgcorrigidos, total_bgcorrigidos],
                     marker=dict(color=['green', 'red']),
                     text=[f'Média: {media_bgcorrigidos:.2f} bgcorrigidos', f'Total: {total_bgcorrigidos:.2f} bgcorrigidos'],
                     textposition='auto'
                    ))
# Personalizar o layout do gráfico
fig.update_layout(title='Média e Total de Bugs Corrigidos por Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Estatísticas',
                  yaxis_title='Bugs Corrigidos',
                  showlegend=False,
                  width=600  # Definindo a largura do gráfico para 600 pixels
                  )
# Mostrar o gráfico interativo
fig.show()

--------------------------------------------------------------------------------

- Total de Tarefas Concluídas
- Média Diária de Tarefas Concluídas

In [45]:
# Criar DataFrame a partir das tabelas
tconcluidas = pd.DataFrame(tabelas)

# Calcular a média e a soma total dos Tarefas Concluídas
media_tconcluidas = horas['Tarefas Concluídas'].mean()
total_tconcluidas = horas['Tarefas Concluídas'].sum()

# Criar figura com os dados
fig = go.Figure()

# Adicionar as barras para média e total e configuralas
fig.add_trace(go.Bar(x=['Média Diária', 'Total por Semana'],
                     y=[media_tconcluidas, total_tconcluidas],
                     marker=dict(color=['green', 'red']),
                     text=[f'Média: {media_tconcluidas:.2f} tconcluidas', f'Total: {total_tconcluidas:.2f} tconcluidas'],
                     textposition='auto'
                    ))
# Personalizar o layout do gráfico
fig.update_layout(title='Média e Total de Tarefas Concluídas por Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Estatísticas',
                  yaxis_title='Tarefas Concluídas',
                  showlegend=False,
                  width=600  # Definindo a largura do gráfico para 600 pixels
                  )
# Mostrar o gráfico interativo
fig.show()

--------------------------------------------------------------------------------

- Produtividade Diária (Tarefas Concluídas por Hora)

In [46]:
# Criar DataFrame a partir das tabelas
dftabelas = pd.DataFrame(tabelas)

# Calcular a produtividade diária (tarefas concluídas por hora) para cada dia da semana
dftabelas['Produtividade Diária'] = (dftabelas['Tarefas Concluídas'] / dftabelas['Horas Trabalhadas']).round(2)

# Definir o dia da semana em português
days_of_week = {
    'Segunda': 'Segunda-feira',
    'Terça': 'Terça-feira',
    'Quarta': 'Quarta-feira',
    'Quinta': 'Quinta-feira',
    'Sexta': 'Sexta-feira',
    'Sábado': 'Sábado',
    'Domingo': 'Domingo'
}
dftabelas['Dia da Semana'] = dftabelas['Dia'].map(days_of_week)

# Criar o gráfico de barras com o Plotly
fig = px.bar(dftabelas, x='Dia da Semana', y='Produtividade Diária', text='Produtividade Diária',
             hover_data={'Horas Trabalhadas': True, 'Tarefas Concluídas': True},
             labels={'Produtividade Diária': 'Produtividade Diária (Tarefas Concluídas por Hora)'},
             color_discrete_sequence=px.colors.qualitative.Set2)

# Personalizar o layout do gráfico
fig.update_layout(title='Produtividade Diária ao Longo da Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Dia da Semana',
                  yaxis_title='Produtividade Diária (Tarefas Concluídas por Hora)',
                  xaxis={'categoryorder': 'array', 'categoryarray': list(days_of_week.values())},
                  xaxis_tickangle=-45,
                  barmode='group',
                  bargap=0.2,
                  bargroupgap=0.1)

# Mostrar o gráfico interativo
fig.show()

In [47]:
# Criar DataFrame a partir das tabelas
dftabelas = pd.DataFrame(tabelas)

# Calcular os Bugs Corrigidos por Hora para cada dia da semana
dftabelas['Bugs Corrigidos por Hora'] = (dftabelas['Bugs Corrigidos'] / dftabelas['Horas Trabalhadas']).round(2)

# Definir o dia da semana em português
days_of_week = {
    'Segunda': 'Segunda-feira',
    'Terça': 'Terça-feira',
    'Quarta': 'Quarta-feira',
    'Quinta': 'Quinta-feira',
    'Sexta': 'Sexta-feira',
    'Sábado': 'Sábado',
    'Domingo': 'Domingo'
}
dftabelas['Dia da Semana'] = dftabelas['Dia'].map(days_of_week)

# Criar o gráfico de barras com o Plotly
fig = px.bar(dftabelas, x='Dia da Semana', y='Bugs Corrigidos por Hora', text='Bugs Corrigidos por Hora',
             hover_data={'Horas Trabalhadas': True, 'Bugs Corrigidos': True},
             labels={'Bugs Corrigidos por Hora': 'Bugs Corrigidos (Bugs Corrigidos por Hora)'},
             color_discrete_sequence=px.colors.qualitative.Set2)

# Personalizar o layout do gráfico
fig.update_layout(title='Bugs Corrigidos por Hora ao Longo da Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Dia da Semana',
                  yaxis_title='Bugs Corrigidos (Bugs Corrigidos por Hora)',
                  xaxis={'categoryorder': 'array', 'categoryarray': list(days_of_week.values())},
                  xaxis_tickangle=-45,
                  barmode='group',
                  bargap=0.2,
                  bargroupgap=0.1)

# Mostrar o gráfico interativo
fig.show()

## Relatório
- Tivemos uma média diária de 6.14 horas trabalhadas com uma média de 4.14 terefaz concluídas e a média de 2.29 bugs corrigidos
- O total de horas trabalhadas por semana e de 43 horas, O total de tarefaz conclídas e de 29 tarefas, E o total de bugs corrigidos e de 16 bugs
- Produtividade Diária (Tarefas Concluídas por Hora) na segunda-feira teve o maior desempenho de tarefaz comcluídas por horas e domingo o menor
- Temos a quatidades de bugs corrigidos por hora e vimos que na quarta feira tivemos o menor números de bugs corrigidos em relação as horas e na quinta-feira o maior números de bugs corrigido em relação as horas

